files/plugin-EnvironmentVariables-4.0.1/EnvironmentVariables.php

```diff
@@ -0,0 +1,17 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\EnvironmentVariables;
+
+class EnvironmentVariables extends \Piwik\Plugin
```

Comment:
It looks like we are including two versions of this plugin (3.0.0 and 4.0.1). Is that intended?

I'm also curious where the actual plugin gets installed? This seems to just be a loader and then we define plugins to install in `plugin.json`... but it's unclear to me if the plugin gets installed during a build phase or only when the container starts running. If the latter, is it installed fresh every reboot? (i.e. might we expect weird issues if plugins are not available during a service restart?)

Reply:
Good catch! The plugin version is linked to the Matomo version. So, for now, we are only using the 3.0.0 version of the EnvironmentVariables plugin in the initial migration to the AWS Organization. I also copied in the 4.0.1 version when I was testing some of the upgrade processes and did not delete it (knowing we would need it in the near future).

The plugin itself is "installed" with this [COPY line](https://github.com/MITLibraries/docker-matomo/blob/d7e2d63853e3c50ce801648040b08c535f49ccec/Dockerfile#L4) in the Dockerfile. 

There is some combination of both the `config.ini.php` file (which lists the installed plugins) and the database itself (which tracks whether an installed plugin is actually activated). 

So, in all of the testing so far, there have been **no** issues with container reboots and losing track of either installed or activated plugins.

Reply:
Gotcha. This plugin structure is really weird as the main method just returns true but I see looking a bit closer that there is also a "config" file that seems to do all the actual work. Ah, PHP. Now that I see what that file is doing, I better understand that yes this is not loading live and is done during the build step so my concern has been addressed. Thanks!